spring-xd-rest-client/src/main/java/org/springframework/xd/rest/client/impl/StreamTemplate.java

```diff
@@ -95,4 +97,17 @@ public void destroyAll() {
 		restTemplate.delete(resources.get("streams/definitions").expand());
 	}
 
+	@Override
+	public void deleteQueue(String name, Map<String, String> properties) {
+		String rootUri = resources.get("streams/clean/rabbit").toString() + "/{name}";
+		UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(rootUri);
+
+		for(Map.Entry<String, String> entry: properties.entrySet()) {
+			builder.queryParam(entry.getKey(), entry.getValue());
+		}
```

Comment:
Will need to introduce an intermediary class, as `AbstractTemplate` is really just the base class for REST support.